### PR TITLE
GH-771 TABLE (AUTO CALC ITEMS PER PAGE) AND PAGINATOR UPDATE

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @dtsnell4 Please Review

Adds new publish parameter autoCalcIPP
Changes Paginator so that its fixed height
Adds new features to Paginator (jump to page)

Fixes GH-771

### CODE REVIEW

**Comment by mlzummo** [comment]
http://rawgit.com/mlzummo/Visualization/GH-771_TABLE_PAGINATOR_UPDATE/demos/dermatology.html?src/other/Table

**Comment by GordonSmith** [comment]
You need to set the min and max permissible values in the number input field.

File: src/other/Paginator.js [hunk 1]
```diff
@@ -16,15 +16,43 @@
...
+
+        this.side.append("span")
+            .classed("side", true)
+            .text("Page ")
+        ;
+
+        this.side.append("input")
+            .attr("type","number")
+            .attr("class","currentPageNumber")
+            .attr("value",1)
```
Comment (GordonSmith) [inline src/other/Paginator.js]: Should be ".property" and not ".attr"

File: src/other/Paginator.js [hunk 2]
```diff
@@ -35,41 +63,66 @@
...
+                if (this.pageNumber() + x > 0 && this.pageNumber() + x <= this._tNumPages) {
+                    this._numList.push(this.pageNumber() + x);
+                }
             }
-            this._numList.push("next");
+            this._numList.push("last");
         }
 
+        this.side.select(".total").text(" of "+this._tNumPages);
+        this.side.select(".currentPageNumber").attr("value",this.pageNumber());
```
Comment (GordonSmith) [inline src/other/Paginator.js]: should be .property and not .attr

File: src/other/Paginator.js [hunk 3]
```diff
@@ -35,41 +63,66 @@
...
+                        span.innerHTML = d;
+
+                        li.appendChild(span);
+
+                    return li;
+                }
+            })
+            .on("click", function(d, i) {
+                d3.event.preventDefault();
+                context.side.select(".currentPageNumber").attr("value",context.pageNumber());
```
Comment (GordonSmith) [inline src/other/Paginator.js]: Should be .property and not .attr

**Comment by mlzummo** [comment]
@GordonSmith Fixed

**Pull request merged**